File: VB/ControlPatcherRegistrator.vb
Imports DevExpress.XtraEditors
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.ComponentModel.Design
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms

Namespace WindowsApplication1

    Public Class ControlPatcher(Of T As Control)
        Private _ProcessControlAction As Action(Of T)
        Public Sub New(ByVal processControlAction As Action(Of T))
            _ProcessControlAction = processControlAction
        End Sub

        Public Sub ProcessContainer(ByVal container As Control)
            For Each control As Control In container.Controls
                If control.GetType() Is GetType(T) Then
                    ProcessControl(control)
                Else
                    ProcessContainer(control)
                End If
            Next control
        End Sub

        Private Sub ProcessControl(ByVal control As Object)
            _ProcessControlAction(TryCast(control, T))
        End Sub
    End Class

    Public Class ControlPatcherRegistrator
        Inherits Component

        Private Function IsDesignMode() As Boolean
            Return DesignMode OrElse DevExpress.Utils.Design.DesignTimeTools.IsDesignMode
        End Function

        Public Sub New(ByVal container As IContainer)
            If Not IsDesignMode() Then
                Return
            End If


        End Sub

        Public Property ContainerControl() As Form
            Get
                Return _containerControl
            End Get
            Set(ByVal value As Form)
                _containerControl = value
                SubscribeEvents()
            End Set
        End Property
        Private _containerControl As Form = Nothing

        Public Overrides Property Site() As ISite
            Get
                Return MyBase.Site
            End Get
            Set(ByVal value As ISite)
                MyBase.Site = value
                If value Is Nothing Then
                    Return
                End If
                Dim host As IDesignerHost = TryCast(value.GetService(GetType(IDesignerHost)), IDesignerHost)
                If host Is Nothing Then
                    Return
                End If
                Dim componentHost As IComponent = host.RootComponent
                If TypeOf componentHost Is Form Then
                    _containerControl = TryCast(componentHost, Form)
                End If
            End Set
        End Property

        Private Sub SubscribeEvents()
            If ContainerControl Is Nothing Then
                Return
            End If
            RemoveHandler ContainerControl.Load, AddressOf ContainerControl_Load
            AddHandler ContainerControl.Load, AddressOf ContainerControl_Load
        End Sub

        Private Sub ContainerControl_Load(ByVal sender As Object, ByVal e As EventArgs)
            Register()
        End Sub

        Public Sub Register()
            CType(New ControlPatcher(Of ButtonEdit)(Sub(edit) edit.Properties.TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor), ControlPatcher(Of ButtonEdit)).ProcessContainer(ContainerControl)
            CType(New ControlPatcher(Of DateEdit)(Sub(edit) edit.Properties.WeekDayAbbreviationLength = 3), ControlPatcher(Of DateEdit)).ProcessContainer(ContainerControl)
            CType(New ControlPatcher(Of TextEdit)(Sub(edit) edit.Properties.NullValuePrompt = "Enter your text here"), ControlPatcher(Of TextEdit)).ProcessContainer(ContainerControl)
        End Sub
    End Class
End Namespace

File: CS/ControlPatcherRegistrator.cs
using DevExpress.XtraEditors;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.ComponentModel.Design;
using System.Linq;
using System.Text;
using System.Windows.Forms;

namespace WindowsApplication1
{

    public class ControlPatcher<T> where T : Control
    {
        Action<T> _ProcessControlAction;
        public ControlPatcher(Action<T> processControlAction)
        {
            _ProcessControlAction = processControlAction;
        }

        public void ProcessContainer(Control container)
        {
            foreach (Control control in container.Controls)
            {
                if (control.GetType() == typeof(T))
                {
                    ProcessControl(control);
                }
                else
                    ProcessContainer(control);
            }
        }

        private void ProcessControl(object control)
        {
            _ProcessControlAction(control as T);
        }
    }

    public class ControlPatcherRegistrator : Component
    {

        bool IsDesignMode()
        {
            return DesignMode || DevExpress.Utils.Design.DesignTimeTools.IsDesignMode;
        }

        public ControlPatcherRegistrator(IContainer container)
        {
            if (!IsDesignMode())
                return;


        }

        public Form ContainerControl
        {
            get { return _containerControl; }
            set
            {
                _containerControl = value;
                SubscribeEvents();
            }
        }
        private Form _containerControl = null;

        public override ISite Site
        {
            get { return base.Site; }
            set
            {
                base.Site = value;
                if (value == null)
                    return;
                IDesignerHost host = value.GetService(typeof(IDesignerHost)) as IDesignerHost;
                if (host == null)
                    return;
                IComponent componentHost = host.RootComponent;
                if (componentHost is Form)
                {
                    _containerControl = componentHost as Form;
                }
            }
        }

        void SubscribeEvents()
        {
            if (ContainerControl == null)
                return;
            ContainerControl.Load -= ContainerControl_Load;
            ContainerControl.Load += ContainerControl_Load;
        }

        private void ContainerControl_Load(object sender, EventArgs e)
        {
            Register();
        }

        public void Register()
        {
            new ControlPatcher<ButtonEdit>((edit) => edit.Properties.TextEditStyle = DevExpress.XtraEditors.Controls.TextEditStyles.DisableTextEditor).ProcessContainer(ContainerControl);
            new ControlPatcher<DateEdit>((edit) => edit.Properties.WeekDayAbbreviationLength = 3).ProcessContainer(ContainerControl);
            new ControlPatcher<TextEdit>((edit) => edit.Properties.NullValuePrompt = "Enter your text here").ProcessContainer(ContainerControl);
        }
    }
}
